### PR TITLE
VideoPress: enqueue token bridge script in the block editor context

### LINKED ISSUE #27020: VideoPress: Private videos are not visualized in the Block Editor context
Private videos don't load in `v6`.

Image

* Upload a video with VideoPress
* Set private -> private
* Go to block editor
* Create a VideoPress Video block (v6)
* Use the uploaded video

### PR DESCRIPTION
**Pull request by retrofox**
Fixes https://github.com/Automattic/jetpack/issues/27020

#### Changes proposed in this Pull Request:

* VideoPress: enqueue token bridge script when `init` action

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Create a video VideoPress video
* Set privacy -> private
* Go to block editor
* Create a new VideoPress Video v6 block
* Use the private video
* Confirm, without these changes, the video is not visualized, getting the `Post message time out`
* Confirm with these changes the video is visualized

before | after
------|------
 | image

### CODE REVIEW

**Review by nunyvega — DISMISSED** [review]
Works well! 

For learning purposes:
 why do you choose `init` over `wp_loaded` in this case?

**Comment by retrofox** [comment]
> For learning purposes: why do you choose `init` over `wp_loaded` in this case?

Good question and we should probably change it :-/. I opted for `init` because we need to guarantee the file availability as soon as possible since it's required when the script runs.
Also, we need to expose it for both contexts: wp-admin and Block Editor.

I was considering using [enqueue_block_editor_assets](https://developer.wordpress.org/reference/hooks/enqueue_block_editor_assets/), but then we'll have to hook the action twice: one for the wp-admin and another one for the editor.

That's fine. Let me update it, and then we can test again.

**Review by nunyvega — APPROVED** [review]
Works as expected! my only question here would be if this approach also works outside of the post editor, like in a p2 front-end post creator or some kind of third party editor. 

Regardless of that, this approach fixes our current issue, and the other ones can be contemplated in a different PR, in case that's needed.

**Review by nunyvega — DISMISSED** [review]
I tested it in the Elementor environment, and it fails with this approach. 

The `init` approach works, so let's favour that one

File: projects/packages/videopress/src/class-jwt-token-bridge.php [hunk 1]
```diff
@@ -32,8 +32,7 @@ public static function init() {
 			return;
 		}
 
-		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );
-
+		add_action( 'init', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );
```
Comment (nunyvega) [inline projects/packages/videopress/src/class-jwt-token-bridge.php]:
```suggestion
		// Expose the VideoPress token to the Block Editor context, including the front-end editor.
		add_action( 'enqueue_block_assets', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );
		add_action( 'enqueue_block_editor_assets', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );

		// Expose the VideoPress token to the WPAdmin context.
		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );
```

This change would make it work on the front-end editor and third-party editors too.

**Review by dhasilva — APPROVED** [review]
Works as expected here

**Pull request merged**